src/main/java/seedu/address/model/patient/ICNumber.java

```diff
@@ -0,0 +1,42 @@
+package seedu.address.model.patient;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+public class ICNumber {
+    public static final String MESSAGE_CONSTRAINTS =
+            "IC Number should start and end with an alphabet with non negative numbers in between";
+    public static final String VALIDATION_REGEX = "^[A-Z]\\d{7}[A-Z]$";
+    public final String value;
+
+    public ICNumber(String IC) {
```

Comment:
Hi Riya, referencing from the Java Coding Standard, i think ICNumber should be IcNumber instead since abbreviation should not be uppercase. 

Could refer to this link!!
[Coding standard] https://se-education.org/guides/conventions/java/intermediate.html#naming